operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourceEventFilters.java

```diff
@@ -0,0 +1,102 @@
+package io.javaoperatorsdk.operator.processing.event.internal;
+
+import io.fabric8.kubernetes.client.CustomResource;
+
+/**
+ * Convenience implementations of, and utility methods for, {@link CustomResourceEventFilter}.
+ */
+public final class CustomResourceEventFilters {
+
+  private static final CustomResourceEventFilter<CustomResource> USE_FINALIZER =
```

Comment:
Again, this won't work in general. We won't have the old resource in all cases in the future. 
We have to clarify this, we might end up using actually reflectors instead of informers (or just ListWatcher?). So I'm not sure about this interface cotaining the oldObject.

Reply:
For example this is why we store the lastGenerationProcessedSuccessfully.

Reply:
To be honest, I'm not sure if the operator should cache the last generation as it should be the controller that has the ultimate responsibility to define when it has successfully processed the latest generation (i.e. based on `.status.observedGeneration` field lot of controller are adding these days).

This is also one of the reason for which I think it would be very nice to add the option to define custom predicates.

Reply:
I agree. With the new caching approach this should go away too, and have it in status.

Just maybe to think about this to do it seamlessly, so how the framework could do for the user transparently, with minimal effort. Best would be to create an issue for it, where we discuss the possibilities. (Like if there is an observedGeneration field in status we handle it automatically, or such)

Reply:
Thinking about this aloud, I wonder if it would be better to move the filtering logic to the controller itself in the form of an interfaces, like:

```java
interface ResourceController<T extends CustomResource<Spec, Status>> {
    default boolean process(T resource) {
        return true;
    }
}
```

Then we can add some special `ResourceController` interface, like

```java
interface GenerationAwareResourceController<T extends GenerationAwareCustomResource<Spec, Status>> {
    @Override
    default boolean process(T resource) {
        return resource.status.observedGeneration < resource.meta.generation ;
    }
}
```